src/app/app.component.spec.ts

```diff
@@ -1,81 +1,106 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ComponentFixture, TestBed, ComponentFixtureAutoDetect } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
+import { BrowserModule, By } from '@angular/platform-browser';
 import { AppComponent } from './app.component';
-import { BrowseComponent } from './browse/browse.component';
-import { DashboardComponent } from './dashboard/dashboard.component';
+import { testRoutes, RouterLinkDirectiveStub } from './test-helpers';
+import { DebugElement, NO_ERRORS_SCHEMA } from '@angular/core';
 import { HeaderComponent } from './header/header.component';
 import { NavBarComponent } from './nav-bar/nav-bar.component';
-import { OrgStatComponent } from './org-stat/org-stat.component';
-import { RepoPreviewComponent } from './repo-preview/repo-preview.component';
+import { RepoDetailComponent } from './repo-detail/repo-detail.component';
+import { SearchComponent } from './search/search.component';
+import { HomeComponent } from './home/home.component';
+
 
 describe('AppComponent', () => {
   let app: AppComponent;
   let fixture: ComponentFixture<AppComponent>;
+  let compiled: HTMLElement;
+  let linkElements: DebugElement[];
+  let routerLinks: RouterLinkDirectiveStub[];
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       imports: [
-        RouterTestingModule
+        RouterTestingModule.withRoutes(testRoutes),
+        BrowserModule
       ],
       declarations: [
         AppComponent,
         HeaderComponent,
         NavBarComponent,
-        DashboardComponent,
-        BrowseComponent,
-        OrgStatComponent,
-        RepoPreviewComponent
+        RouterLinkDirectiveStub,
+        HomeComponent,
+        SearchComponent,
+        RepoDetailComponent
+      ],
```

Comment:
This is really cool. Looks like more of an integration style test to me. Potential downside is that the test depends on so many components (and potentially the services those components depend on). I think it's fine to keep it, but just pointing it out since we haven't written tests like this yet on the team.

Reply:
The Angular site actually suggested stubbing out these components so that you wouldn't have to declare every service or component each of these required. But decided to skip that step since these tests ran fine without declaring dependencies

Curious to find out how the Shure team would test routing! These tests were super difficult for me to figure out, and would be happy to learn a better way

Reply:
I think that if there's a link in a component to a different route, we just spy on the router.navigate function.
Most of the links are in the tabs though, and those components (TabComponent, TabGroupComponent) are general enough that we can write router tests using dummy components instead of the real ones that have tons of dependencies.

Reply:
Got it! Will make a new ticket for updating these router tests to use a spy